go-controller/pkg/node/vrfmanager/vrf_manager.go

```diff
@@ -0,0 +1,346 @@
+package vrfmanager
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	"github.com/vishvananda/netlink"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+)
+
+type vrf struct {
+	name               string
+	table              uint32
+	enslavedInterfaces sets.Set[string]
+	deleted            bool
+}
+
+type Controller struct {
+	mu   *sync.Mutex
+	vrfs map[string]vrf
```

Comment:
ok, but still AddVRF and DeleteVRF still depends on VRF name for addition and deletion, hope nodes would have required udev rules to keep the same device upon reboots.

Reply:
upon reboots we are ok because we need to rebuild the cache but i am more worried about interfaces changing names during runtime which is possible and supported.
Can you maybe do this as a follow up to this PR? Long term its better I think to use ifindex.